File: components/credentials/src/main/java/org/cloudfoundry/credhub/audit/entities/UpdateTransitionalVersion.java
package org.cloudfoundry.credhub.audit.entities;

import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

public class UpdateTransitionalVersion implements RequestDetails {
  private String version;


  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.UPDATE_TRANSITIONAL_VERSION;
  }

  public String getVersion() {
    return version;
  }

  public void setVersion(final String version) {
    this.version = version;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/UpdateTransitionalVersion.kt
package org.cloudfoundry.credhub.audit.entities

import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails

class UpdateTransitionalVersion : RequestDetails {
    var version: String? = null

    override fun operation(): OperationDeviceAction {
        return OperationDeviceAction.UPDATE_TRANSITIONAL_VERSION
    }
}
